src/cmd/api/router/router.go

```diff
@@ -53,17 +56,19 @@ type responseWriter struct {
 func NewRouter(config config.Config, clients *Clients, uuidGen *UUIDGenerator, DAOs *DAOs) http.Handler {
 	metrics.InitMetrics()
 	r := mux.NewRouter()
+	s := r.PathPrefix("/api").Subrouter()
 	r.Use(prometheusMiddleware)
-	r.Use(httpauth.SimpleBasicAuth(config.UserAuth, config.PwdAuth))
-
-	r.PathPrefix("/api/v1/videos/{id}/streams/master.m3u8").Handler(controllers.VideoGetMasterHandler{S3Client: clients.S3Client, UUIDGen: uuidGen.UUIDGen}).Methods("GET")
-	r.PathPrefix("/api/v1/videos/{id}/streams/{quality}/{filename}").Handler(controllers.VideoGetSubPartHandler{S3Client: clients.S3Client, UUIDGen: uuidGen.UUIDGen}).Methods("GET")
-	r.PathPrefix("/api/v1/videos/list/{attribute}/{order}/{page}/{limit}").Handler(controllers.VideosListHandler{VideosDAO: &DAOs.VideosDAO}).Methods("GET")
-	r.PathPrefix("/api/v1/videos/{id}/delete").Handler(controllers.VideoDeleteVideoHandler{S3Client: clients.S3Client, VideosDAO: &DAOs.VideosDAO, UploadsDAO: &DAOs.UploadsDAO, UUIDGen: uuidGen.UUIDGen}).Methods("DELETE")
-	r.PathPrefix("/api/v1/videos/{id}/info").Handler(controllers.VideoGetInfoHandler{VideosDAO: &DAOs.VideosDAO, UUIDGen: uuidGen.UUIDGen}).Methods("GET")
-	r.PathPrefix("/api/v1/videos/list").Handler(controllers.VideosListHandler{VideosDAO: &DAOs.VideosDAO}).Methods("GET")
-	r.PathPrefix("/api/v1/videos/upload").Handler(controllers.VideoUploadHandler{S3Client: clients.S3Client, AmqpClient: clients.AmqpClient, VideosDAO: &DAOs.VideosDAO, UploadsDAO: &DAOs.UploadsDAO, UUIDGen: uuidGen.UUIDGen}).Methods("POST")
-	r.PathPrefix("/api/v1/videos/{id}/status").Handler(controllers.VideoGetStatusHandler{VideosDAO: &DAOs.VideosDAO, UUIDGen: uuidGen.UUIDGen}).Methods("GET")
+	s.Use(httpauth.SimpleBasicAuth(config.UserAuth, config.PwdAuth))
+
+	r.PathPrefix("/ws").Handler(controllers.WSHandler{}).Methods("GET")
```

Comment:
Not a big fan of the general organization. We can rename s => v1 and change the path from /api => /api/v1.

And we should regroup all the subrouter r together then all the s/v1.

Reply:
The correction is done.